tests/codegen/llvm/optional_positional_parameter.ll

```diff
@@ -37,12 +37,7 @@ entry:
   %cpu.id.bounded = and i64 %get_cpu_id, %1
   %2 = getelementptr [1 x [1 x [64 x i8]]], ptr @get_str_buf, i64 0, i64 %cpu.id.bounded, i64 0, i64 0
   call void @llvm.memset.p0.i64(ptr align 1 %2, i8 0, i64 64, i1 false)
-  call void @llvm.lifetime.start.p0(i64 -1, ptr %str)
-  call void @llvm.memset.p0.i64(ptr align 1 %str, i8 0, i64 1, i1 false)
-  store [1 x i8] zeroinitializer, ptr %str, align 1
-  %3 = ptrtoint ptr %str to i64
-  %probe_read_kernel_str = call i64 inttoptr (i64 115 to ptr)(ptr %2, i32 64, i64 %3)
-  call void @llvm.lifetime.end.p0(i64 -1, ptr %str)
+  %probe_read_kernel_str = call i64 inttoptr (i64 115 to ptr)(ptr %2, i32 64, i64 ptrtoint (ptr @0 to i64))
```

Comment:
depending on how deep you wanna go on surgical optimizations, but seems like probe_read_kernel_str should not be necessary here. should be able to do memcpy

Reply:
Yeah, good point. Let me look into that for another PR.